app/controllers/api/v1/credentials_controller.rb

```diff
@@ -3,7 +3,16 @@ class CredentialsController < ApiController
     respond_to :json
 
     def show
-      render json: CredentialsSerializer.new(current_resource_owner).call
+      render json: credentials_serializer.call
```

Comment:
I think we need to refactor the serializer code - it's calling `to_json` internally which isn't the right way of doing it (in fact `to_json` isn't implemented by Rails). It should just return a hash to the `render json:` call and that will call `as_json` and then `ActiveSupport::JSON.encode` which is the correct way of doing things.

As part of that we may want to think about using [Active Model Serializers][1] and rewrite this block using `respond_to`, e.g:

``` ruby
def show
  respond_to do |format|
    format.json { render json: credentials }
  end
end

private

def credentials
  Credentials.new(user: current_resource_owner, application: doorkeeper_token.application)
end
```

This will give us proper 422 errors when someone tries to do a non-JSON request and takes at lot of the plumbing out of the controller and just into the serializer classes.

[1]: https://github.com/rails-api/active_model_serializers/tree/v0.9.3